src/main/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecution.java

```diff
@@ -737,12 +746,16 @@ public void onLoad(FlowExecutionOwner owner) throws IOException {
                 }
             } else {  // See if we can/should resume build
                 if (canResume()) {
+                    // Before we resume, we need to unset persistedClean in case Jenkins restarts again.
+                    persistedClean = null;
+                    saveOwner();
```

Comment:
Should add a test case for this. The test would use a PERFORMANCE_OPTIMIZED Pipeline, restart Jenkins normally, then restart Jenkins again with a hard shutdown and make sure we get the error message that the Pipeline can't be resumed rather than a raw exception.

Reply:
I came up with a test case, but it doesn't fail without this line because of a random save that happens when `CpsFlowExecution.notifyListeners` is called. IIUC, https://github.com/jenkinsci/workflow-cps-plugin/pull/234 was trying to stop that save from occurring for PERFORMANCE_OPTIMIZED Pipelines, so I will look into making related changes in this PR.